# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/BlockReflowState.cpp
# Commit: 0fd4431f5279
# Full Hash: 0fd4431f527935234d764a167cc5a20acfc323eb
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2023-09-28 21:51:27
# Regressor Bug: 1731541
# File Overlap Count: 1
# Description:
#   Bug 1731541 - Implement text-wrap: balance for nsBlockFrame reflow. r=emilio
#   
#   A simple form of balance for short blocks, implemented by incrementally reducing
#   the effective inline-size used during line-breaking, up to the point where an
#   extra line would be created.
# ==============================================================================

diff -r 5926241957b4 -r 0fd4431f5279 layout/generic/BlockReflowState.cpp
--- a/layout/generic/BlockReflowState.cpp	Thu Sep 28 08:31:20 2023 +0000
+++ b/layout/generic/BlockReflowState.cpp	Thu Sep 28 08:31:20 2023 +0000
@@ -27,17 +27,16 @@
 using namespace mozilla;
 using namespace mozilla::layout;
 
-BlockReflowState::BlockReflowState(const ReflowInput& aReflowInput,
-                                   nsPresContext* aPresContext,
-                                   nsBlockFrame* aFrame, bool aBStartMarginRoot,
-                                   bool aBEndMarginRoot,
-                                   bool aBlockNeedsFloatManager,
-                                   const nscoord aConsumedBSize,
-                                   const nscoord aEffectiveContentBoxBSize)
+BlockReflowState::BlockReflowState(
+    const ReflowInput& aReflowInput, nsPresContext* aPresContext,
+    nsBlockFrame* aFrame, bool aBStartMarginRoot, bool aBEndMarginRoot,
+    bool aBlockNeedsFloatManager, const nscoord aConsumedBSize,
+    const nscoord aEffectiveContentBoxBSize, const nscoord aInset)
     : mBlock(aFrame),
       mPresContext(aPresContext),
       mReflowInput(aReflowInput),
       mContentArea(aReflowInput.GetWritingMode()),
+      mInsetForBalance(aInset),
       mPushedFloats(nullptr),
       mOverflowTracker(nullptr),
       mBorderPadding(